nukebuild/Build.cs

```diff
@@ -134,97 +132,84 @@ void ExecWait(string preamble, string command, string args)
                 );
 
             else
-                DotNetBuild(Parameters.MSBuildSolution, c => c
+                DotNetBuild(c => c
+                    .SetProjectFile(Parameters.MSBuildSolution)
                     .AddProperty("PackageVersion", Parameters.Version)
                     .SetConfiguration(Parameters.Configuration)
                 );
         });
-    
-    void RunCoreTest(string project)
+
+    void RunCoreTest(string projectName)
     {
-        if(!project.EndsWith(".csproj"))
-            project = System.IO.Path.Combine(project, System.IO.Path.GetFileName(project)+".csproj");
-        Information("Running tests from " + project);
-        XDocument xdoc;
-        using (var s = File.OpenRead(project))
-            xdoc = XDocument.Load(s);
-
-        List<string> frameworks = null;
-        var targets = xdoc.Root.Descendants("TargetFrameworks").FirstOrDefault();
-        if (targets != null)
-            frameworks = targets.Value.Split(';').Where(f => !string.IsNullOrWhiteSpace(f)).ToList();
-        else 
-            frameworks = new List<string> {xdoc.Root.Descendants("TargetFramework").First().Value};
-        
-        foreach(var fw in frameworks)
+        Information($"Running tests from {projectName}");
+        var project = Solution.GetProject(projectName).NotNull("project != null");
+
+        foreach (var fw in project.GetTargetFrameworks())
```

Comment:
Does it use MSBuild evaluation or parses stuff manually? We've seen issues with MSBuild on not-quite-supported OSes

Reply:
It uses MSBuild for evaluation. Could you try on those _not-quite-supported_ environments?